tracing/error_reporter_hooks.go

```diff
@@ -0,0 +1,44 @@
+package tracing
+
+import "github.com/reddit/baseplate.go/log"
+
+const (
+	success = "success"
+	fail    = "fail"
+)
+
+// ErrorReporterBaseplateHook registers each Server Span with an
+// ErrorReporterSpanHook that will publish errors sent to OnEnd to Sentry.
+type ErrorReporterBaseplateHook struct{}
+
+// OnServerSpanCreate registers SentrySpanHook on a Server Span.
+func (h ErrorReporterBaseplateHook) OnServerSpanCreate(span *Span) error {
+	span.RegisterHook(ErrorReporterSpanHook{})
+	return nil
+}
+
+// ErrorReporterSpanHook publishes errors sent to OnEnd to Sentry.
+type ErrorReporterSpanHook struct{}
+
+// OnCreateChild is a nop.
+func (h ErrorReporterSpanHook) OnCreateChild(child *Span) error {
+	return nil
+}
+
+// OnStart is a nop
+func (h ErrorReporterSpanHook) OnStart(span *Span) error {
+	return nil
+}
+
+// OnEnd logs a message and sends err to Sentry if err is non-nil.
+func (h ErrorReporterSpanHook) OnEnd(span *Span, err error) error {
+	if err != nil {
+		log.ErrorWithRaven("Span ended with an error", err)
```

Comment:
🔕 `log.Error*` will also print call stacks, but call stacks from `hook.OnEnd` might not be that useful. We might want to omit the logging part and only send to sentry here instead?

But if we do want to keep the logging part, we might want to also add span info here, so you can add either `"span", *span` or just `"spanname", span.Name` (as that's the only exported field of `Span`) to the args of `ErrorWithRaven`. You can also implement `Span.String()` so that adding `"span", *span` will output something more useful to the log.